type-helpers.ts

```diff
@@ -0,0 +1,10 @@
+export type PrettyPrint<T> = T extends infer U extends object
```

Comment:
What does `PrettyPrint` do exactly?

Reply:
Given a type such as 
```ts
# type User = {
  name: string
  age: number
}

type Address = {
  street: string
  city: string
  state: string
  zip: number
  country: string
}

type UserWithAddress = User & Address

const me: User = {...}
```

the tooltip when hovering over `me` will show `User & Address`.

However, with the following:

```
type UserWithAddress = PrettyPrint<User & Address>

const me: User = {...}
```

The tooltip would instead show:

```
{
  name: string
  age: number
  street: string
  city: string
  state: string
  zip: number
  country: string
}
```

Reply:
Very nice!  That has always kind of bugged me.

Reply:
Also for those situations satisfies operator of typescript can help without 3rd party deps :D